doc/topics/variables-extending-and-substitutions.rst

```diff
@@ -0,0 +1,274 @@
+or they might both extend a common base configuration.  Of course, cycles
+are not allowed.
+
+Configurations may be named with URLs in the ``extends`` option, in
+which case they may be downloaded from remote servers.  See :ref:`The
+extends-cache buildout option <extends-cache-buildout-option>`.
+
+When a relative path is used in an extends option, it's interpreted
+relative to the path of the extending configuration.
+
+.. _user-default-configuration:
+
+User-default configuration
+==============================
+
+A per-user default configuration may be defined in the ``default.cfg``
+file in the ``.buildout`` subdirectory of a user's home directory
+(``~/.buildout/default.cfg`` on Mac OS and Linux).  This configuration
+is typically used to set up a shared egg or cache directory, as in:
+
+.. code-block:: ini
+
+  [buildout]
+  eggs-directory = ~/.buildout/eggs
+  download-cache = ~/.buildout/download-cache
+  abi-tag-eggs = true
+
+See the section on :doc:`optimizing buildouts with shared eggs and
+download caches <topics/optimizing>` for an explanation of the options
+used in the example above.
+
+.. _merge-values-with-existing-values:
+
+Merging, rather tha overriding values
+=====================================
+
+Normally, values in extending configurations override values in
+extended configurations by replacing them, but it's also possible to
+augment or trim overridden values.  If ``+=`` is used rather than
+``=``, the overriding option value is appended to the original. So,
+for example if we have a base configuration, ``buildout.cfg``:
+
+.. code-block:: ini
+
+   [buildout]
+   parts =
+     py
+     test
+     server
+   ...
+
+And a production configuration ``prod.cfg``, we can add another part,
+``monitor``, like this:
+
+.. code-block:: ini
+
+   [buildout]
+   extends = buildout.cfg
+   parts += monitor
+   ...
+
+In this example, we didn't have to repeat (or necessarily know) the
+base parts to add the ``monitor`` part.
+
+We can also subtract values using ``-=``, so if we wanted to exclude
+the ``test`` part in production:
+
+.. code-block:: ini
+
+   [buildout]
+   extends = buildout.cfg
+   parts += monitor
+   parts -= test
+   ...
+
+Something to keep in mind is that this works by *lines*.  The ``+=``
+form adds the lines in the new data to the lines of the
+old. Similarly, ``-=`` removes *lines* in the overriding option from the
+original *lines*. This is a bit delicate.  In the example above,
+we were careful to put the base values on separate lines, in
+anticipation of using ``-=``.
+
+Merging values also works with option settings provided via the
+:ref:`buildout command line <buildout-command-line>`.  For example, if
+you want to temporarily use a :ref:`development version
+<python-development-projects>` of another project, you can augment the
+buildout :ref:`develop option <develop-option>` on the command-line
+when running buildout:
+
+.. code-block:: console
+
+   buildout develop+=/path/to/other/project
```

Comment:
Added.

I also added as small section on unpinning to the getting-started doc.